### PR TITLE
modified gifti_surface_wf to remove the fsnative transform and replace using to_scanner

### PR DESCRIPTION
**Pull request by ericfeczko**
this pull request address issue #289 -- enabling smriprep to be used with expected inputs for nibabies. Should also improve alignments in edge cases for non-infant processing :)

### CODE REVIEW

File: smriprep/workflows/surfaces.py [hunk 1]
```diff
@@ -531,9 +529,21 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+    surfmorph_list = pe.Node(
+        niu.Merge(3,ravel_inputs=True),
+        name="surfmorph_list",
+        run_without_submitting=True,
+    )
+    fscurv2funcgii = pe.MapNode(
+        fs.MRIsConvert(out_datatype="gii",to_scanner=True,scalarcurv_file=True),iterfield="in_file", name="fscurv2funcgii"
+    )
+    allsurf_list = pe.Node(
+        niu.merge(7,ravel_inputs=True),
```
Comment (mgxd) [inline smriprep/workflows/surfaces.py:543]: is this a list of lists? if not, ravel_inputs isn't needed
```suggestion
        niu.Merge(7),
```

File: smriprep/workflows/surfaces.py [hunk 2]
```diff
@@ -531,9 +529,21 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
     fs2gii = pe.MapNode(
-        fs.MRIsConvert(out_datatype="gii"), iterfield="in_file", name="fs2gii"
+        fs.MRIsConvert(out_datatype="gii",to_scanner=True), iterfield="in_file", name="fs2gii"
+    )
+    surfmorph_list = pe.Node(
+        niu.Merge(3,ravel_inputs=True),
+        name="surfmorph_list",
+        run_without_submitting=True,
+    )
+    fscurv2funcgii = pe.MapNode(
```
Comment (mgxd) [inline smriprep/workflows/surfaces.py:539]: i'm confused about this step - we aren't doing any functional processing here

File: smriprep/workflows/surfaces.py [hunk 3]
```diff
@@ -531,9 +529,21 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+        fs.MRIsConvert(out_datatype="gii",to_scanner=True,scalarcurv_file=True),iterfield="in_file", name="fscurv2funcgii"
```
Comment (mgxd) [inline smriprep/workflows/surfaces.py:540]: the `scalarcurv_file` arg requires an existing file

**Review by mgxd — COMMENTED** [review]
Thanks for the PR @ericfeczko - can you describe the proposed changes a little bit more in depth? I am a little bit confused.

I went through and quickly left some notes - one thing this would greatly benefit from is running a linter and making sure it passes the flake8 checks.

File: smriprep/workflows/surfaces.py (hunk 2 above)
Comment (ericfeczko) [reply smriprep/workflows/surfaces.py:539]: Whoops. This is for outputting the metric surfaces that will be needed for the "goodvoxels" projection fix to eliminate aliasing in the projected outputs.

**Comment by effigies** [comment]
Okay, so I'm reasonably convinced that this is something that needs doing, but we're going to need to continue to use our surface normalization utility, modified to zero out the CRAS metadata without applying it to the coordinates. Here are screenshots of four surfaces viewed in freeview and wbview:

* Red: `mris_convert`
* Green: `mris_convert --to-scanner`
* Blue: `mris_convert` + `NormalizeSurfs` (i.e., add CRAS to points and set CRAS tozero)
* Cyan: `mris_convert --to-scanner` + zero-out CRAS manually in nibabel

###### freeview
freeview
###### wbview
wbview

It's a little hard to see, but basically Blue and Cyan appear correct in both, while red appears correct for freeview and green appears correct for wbview.

I've looked at blue and green, and there is more difference than just the CRAS values (differences can be up to 0.02mm, well above rounding error). So there does seem to be some effect of the obliquity, but I don't understand the transforms involved well enough to reproduce the effect of `--to-scanner` myself.

----

Anyway, my recommendation is to update `NormalizeSurf` to simply zero out the CRAS (we also want to continue adding appropriate metadata to . To keep things isolated, can we separate out the thickness/curv/sulc into a separate PR?

**Comment by effigies** [comment]
Good news: Cyan and green still line up in wbview for infants. Bad news: Nothing lines up for freeview.

infant-freeview
infant-wbview

So we can go ahead with `--to-scanner` + drop CRAS to preserve workbench/FreeSurfer compatibility in reasonably-oriented adults. It's going to take more effort to understand FreeSurfer's interpretation of the two affines to produce GIFTIs that align correctly in both cases.

**Comment by effigies** [comment]
Ah ha! Setting the `VolGeom{X,Y,Z}_{R,A,S}` fields to be `[[-1, 0, 0], [0, 0, 1], [0, -1, 0]]` fixes the cyan to match up with the red in FreeView without damaging it in wbview:

infant-freeview

**Pull request closed without merging**